src/geo/map/cartodb-layer.js

```diff
@@ -100,6 +101,58 @@ var CartoDBLayer = LayerModelBase.extend({
     var meta = this.get('meta');
     var stats = meta && meta.stats;
     return stats && stats.estimatedFeatureCount;
+  },
+
+  getSourceId: function () {
+    var source = this.getSource();
+    return source && source.id;
+  },
+
+  getSource: function () {
+    return this.get('source');
+  },
+
+  setSource: function (source, options) {
+    this.set('source', source, options);
+  },
+
+  /**
+   * Check if an analysis node is the layer's source.
+   * Only torque and cartodb layers have a source otherwise return false.
+   */
+  hasSource: function (analysisModel) {
+    if (!(analysisModel instanceof AnalysisModel)) {
```

Comment:
`CartoDBLayer#equals` already throws a similar error. We should agree on what methods should trigger these kind of errors.

Reply:
* **pros:**  Easy to catch bugs in production, clients will be aware of their errors thanks to a clear trace.

* **cons:** Hard to test, no duck typing for mocks.

IMO we should do a type-check in every method which requires specific parameters

Reply:
> IMO we should do a type-check in every method which requires specific parameters

Every method / function requires specific parameters, right?